### ISSUE TITLE
Wrong sql generated for linq query - table alias is used in wrong subquery

### ISSUE THREAD

**Issue by beercsaba** [linq2db/linq2db#4336]
### Description

At first, i like linq2db very much!

I've created some reusable linq queries, from simpler to complex in class ProductViews.
The complex ones uses the previous ones.
The first 6 works well, but the last 2 (ViewCapacityAndOrderedByPeriod and ViewCapacityUtilization)  don't.
Both of them have the same problem while compiling to sql query: table alias defined in a subquery is used in a outer level of the whole sql query.

Exception message:  42P01: missing FROM-clause entry for table "o_3"  POSITION: 1865
Stack trace:
```   
at Npgsql.Internal.NpgsqlConnector.<<ReadMessage>g__ReadMessageLong|234_0>d.MoveNext()
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Npgsql.NpgsqlDataReader.<NextResult>d__47.MoveNext()
   at Npgsql.NpgsqlDataReader.<NextResult>d__47.MoveNext()
   at Npgsql.NpgsqlDataReader.NextResult()
   at Npgsql.NpgsqlCommand.<ExecuteReader>d__124.MoveNext()
   at Npgsql.NpgsqlCommand.<ExecuteReader>d__124.MoveNext()
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at LinqToDB.Data.DataConnection.ExecuteReader(CommandBehavior commandBehavior)
   at LinqToDB.Data.DataConnection.ExecuteDataReader(CommandBehavior commandBehavior)
   at LinqToDB.Data.DataConnection.ExecuteReader()
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReader()
   at LinqToDB.Linq.QueryRunner.<ExecuteQuery>d__12`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at linq2db_issue.Program.Main(String[] args) in D:\work\playground\BeMyBaker001\peters_repo\linq2db-issue\Program.cs:line 18
```

### Steps to reproduce

From this linq query:
` var q = (from r in dbCtx.Views.Product.ViewCapacityAndOrderedByPeriod() select r).Take(10);`

Where dbCtx.Views.Product.ViewCapacityAndOrderedByPeriod() is like this:
```      
public IQueryable<ViewCapacityAndOrderedByPeriodRow> ViewCapacityAndOrderedByPeriod() =>
         from v1 in dbCtx.Views.Product.ViewCapacityAndOrderedByPeriodAndProduct()
         from v2 in dbCtx.Views.Product.ViewCapacityAndOrderedByPeriodAndProductCategory().Where(o => o.OrderPeriodId == v1.OrderPeriodId && o.ProductCategoryId == v1.CategoryId).DefaultIfEmpty()
         select new ViewCapacityAndOrderedByPeriodRow(v1.OrderPeriodId, v1.ProductId, v1.CategoryId, v1.MaxCapacity, v1.Quantity, v1.FreeCapacity, v2.MaxCapacity, v2.Quantity, v2.FreeCapacity);
```
This sql query is generated: 
```
SELECT
	op_3."Id",
	pop."ProductId",
	o."CategoryId",
	COALESCE(o_1."PeriodOrderLimit",0),
	COALESCE(t3."Quantity",0),
	COALESCE(o_1."PeriodOrderLimit",0) - COALESCE(t3."Quantity",0),
	t7."MaxCapacity",
	t7."Quantity",
	t7."FreeCapacity"
FROM
	"Sales"."OrderPeriod" op_3
		INNER JOIN "Sales"."ProductsPerOrderPeriod" pop ON op_3."Id" = pop."OrderPeriodId"
		LEFT JOIN "Production"."Product" o ON o."Id" = pop."ProductId"
		LEFT JOIN "Production"."ProductCategory" o_1 ON o_1."Id" = o."CategoryId"
		LEFT JOIN (
			SELECT
				COALESCE(Sum(COALESCE(t2."Quantity",0)),0) as "Quantity",
				op."Id",
				t2."ProductId"
			FROM
				"Sales"."OrderPeriod" op
					LEFT JOIN "Sales"."OrderHeader" t1 ON op."Id" = t1."PeriodId"
					LEFT JOIN "Sales"."OrderItem" t2 ON t1."Id" = t2."OrderHeaderId"
			GROUP BY
				op."Id",
				t2."ProductId"
		) t3 ON t3."Id" = op_3."Id" AND t3."ProductId" = pop."ProductId"
		LEFT JOIN (
			SELECT
				COALESCE(o_3."PeriodOrderLimit",0) as "MaxCapacity",
				COALESCE(t6."Quantity",0) as "Quantity",
				COALESCE(o_3."PeriodOrderLimit",0) - COALESCE(t6."Quantity",0) as "FreeCapacity"
			FROM
				(
					SELECT
						op_1."Id",
						vpcc."Id" as "Id_1",
						vpcc."PeriodOrderLimit"
					FROM
						"Sales"."OrderPeriod" op_1,
						"Production"."ProductCategory" vpcc
				) o_3
					LEFT JOIN LATERAL (
						SELECT
							Sum(t5."Quantity") as "Quantity"
						FROM
							"Sales"."OrderPeriod" op_2
								LEFT JOIN "Sales"."OrderHeader" t4 ON op_2."Id" = t4."PeriodId"
								LEFT JOIN "Sales"."OrderItem" t5 ON t4."Id" = t5."OrderHeaderId"
								LEFT JOIN "Production"."Product" o_2 ON o_2."Id" = t5."ProductId"
						WHERE
							op_2."Id" = o_3."Id" AND o_2."CategoryId" = o_3."Id_1"
						GROUP BY
							op_2."Id",
							o_2."CategoryId"
					) t6 ON 1=1
		) t7 ON o_3."Id" = op_3."Id" AND o_3."Id_1" = o."CategoryId"
LIMIT :take
```

Code and database backup (in sql file) can be downloaded from [this url](https://1drv.ms/u/s!AvrRKOGNNWMij-cm3a1O0FQt66sElA?e=OWPvvt)

### Environment details

`Linq To DB` version: *5.3.2*  

Database (with version): *PostgreSql 15* 

ADO.NET Provider (with version): *NpgSql 7.0.6*  (transitive package of linq2db.PostgreSQL 5.3.2)

Operating system: *Windows 10*

.NET Version: *7.0*

**Comment by beercsaba** [comment]
Hi,

I was able to make the solution much smaller. I hope it makes easier to create the unit test for this issue.
You can acces the new smaller version with the [same url](https://1drv.ms/u/s!AvrRKOGNNWMij-cm3a1O0FQt66sElA?e=OWPvvt).
Please use this one!

Best regards,
Csaba

**Comment by MaceWindu** [comment]
Fixed in `6.0.0-preview.1`

**Comment by beercsaba** [comment]
Thank you, @MaceWindu !
